numpy/core/src/common/simd/avx512/operators.h

```diff
@@ -256,4 +256,18 @@
 #define npyv_cmpge_f32(A, B)  _mm512_cmp_ps_mask(A, B, _CMP_GE_OQ)
 #define npyv_cmpge_f64(A, B)  _mm512_cmp_pd_mask(A, B, _CMP_GE_OQ)
 
+// Create mask from the most significant bit of each 8-bit element
+// AVX512F & AVX512BW
+NPY_FINLINE npy_uint64 npyv_movemask_b8(npyv_b8 mask)
+{
+#ifdef NPY_HAVE_AVX512BW_MASK
+    return (npy_uint64)_cvtmask64_u64(mask);
```

Comment:
I'm struggling to match this description to the intel docs, which describe this as `dst := ZeroExtend64(a[63:0])`

Reply:
AVX512 have special registers for mask operations`k0-7`, by using `_cvtmask64_u64()` or just casting directly `(npy_uint64)mask` should generate instruction `KMOVQ`, which move from and to 64-bit mask registers.

Reply:
NOTE: 64-bit mask registers come with CPU features `AVX512BW`(starting from skylake-x)